snippets/cpp/VS_Snippets_Remoting/NCLWebClientUserAgent/CPP/useragent.cpp

```diff
@@ -25,6 +25,7 @@ int main()
    Console::WriteLine( s );
    data->Close();
    reader->Close();
+   delete client;
```

Comment:
I would expect that, if we're creating an object instance via `gcnew`, garbage collector should be responsible for deleting this object and it should cause call on `Dispose` member function as well. Since, I'm not really aware of C++CLI, I have doubts on it.